src/components/draw/DrawDirective.js

```diff
@@ -716,24 +728,12 @@ goog.require('ga_styles_service');
             // Do nothing if the layer does not exist
             return;
           }
-          if (layer.getSource().getFeatures().length == 0) {
-            //if no features to save, delete the file
-            if (layer.adminId) {
-              scope.statusMsgId = '';
-              gaFileStorage.del(layer.adminId).then(function() {
-                layer.adminId = undefined;
-                layer.url = undefined;
-              });
-            }
-            map.removeLayer(layer);
-            return;
-          }
           scope.statusMsgId = 'draw_file_saving';
           var kmlString = gaExportKml.create(layer,
               map.getView().getProjection());
           var id = layer.adminId ||
               gaFileStorage.getFileIdFromFileUrl(layer.url);
-          gaFileStorage.save(id, kmlString,
+          gaFileStorage.save(id, kmlString || '<kml></kml>',
```

Comment:
if kmlString is empty the service save nothing if I remember correctly. And we must have KML valid file to get it back.